doc/release/upcoming_changes/18053.new_feature.rst

```diff
@@ -0,0 +1,4 @@
+``np.einsum`` now accepts arrays with ``object`` dtype
+------------------------------------------------------
+The code path will call python operators on object dtype arrays, much
+like ``np.dot`` and ``np.matmul``.
```

Comment:
There might be an assumption that it is OK to start with `int(0)`, that I am not sure is shared by all code paths (e.g. dot does not).  I can live this this, though, even if it might be considered an issue.

Anyway, have to review the rest carefully, but not today.